### ISSUE TITLE
Attempting to map an object's IEnumerable<X> property to a target List<Y> throws InvalidOperationException

### ISSUE THREAD

**Issue by anaximander23** [agileobjects/AgileMapper#179]
I have an ASP.NET Core API that uses AgileMapper to automatically convert domain objects to DTOs. I do this by having a simple method on the controller whose signature is  `MapResult<TDomain, TDto>(TDomain domainObject)`. Inside this method, I use AgileMapper to map the objects, like so:

```
TDto resultDto = Mapper.Map(domainObject).ToANew<TDto>();
```
I then serialise this and send it back, but this isn't relevant.

In this particular instance, the domain object is a simple POCO that has a property that is a collection of items, along with some other metadata:

```
public class ExampleDomainResultSet
{
    public IEnumerable<DomainResult> Results { get; set; }
    public ISomeMetadata Metadata { get; set; }
}
```

The metadata is used to construct headers for the response, which happens elsewhere; the only thing I need for the response object is the contents of that collection. So, I set up a mapping like this:

```
mapper.WhenMapping
    .From<ExampleDomainResultSet>()
    .To<List<ResultDto>>()
    .Map(x => x.Source.Results.Map().ToANew<List<ResultDto>>())
    .ToTarget();
```

My hope was that this would enable me to call `MapResult<ExampleDomainResultSet, List<ResultDto>>(resultSet)` and just get a `List<ResultDto>` constructed from the contents of that `Results` property.

What I actually get is the following exception:

>System.InvalidOperationException: variable 'rmsToRmsData' of type 'AgileObjects.AgileMapper.ObjectPopulation.IObjectMappingData\`2[System.Collections.Generic.List\`1[MyApp.Dtos.ResultDto],System.Collections.Generic.List`1[MyApp.Dtos.ResultDto]]' referenced from scope '', but it is not defined

[agilemapperexception.txt](https://github.com/agileobjects/AgileMapper/files/3921762/agilemapperexception.txt)

Is this something I'm doing wrong, or is this a problem in AgileMapper?

**Comment by SteveWilkes** [comment]
Hi!

Thanks for reporting this. I'm not sure, but I think it might be the same underlying issue as #174, which is fixed in v1.6-preview5, which is not available [on NuGet](https://www.nuget.org/packages/AgileObjects.AgileMapper).

I've written [a .NET Fiddle](https://dotnetfiddle.net/7tgnlB) demonstrating what you're attempting to do - note that `x.Source.Results.Map().ToANew<List<ResultDto>>()` can just be `x.Source.Results`, as the mapper takes care of mapping the source to the target.

If you still find an issue with v1.6-preview5, please do let me know!

Cheers,

Steve

**Comment by anaximander23** [comment]
Yeah, I tried leaving the mapping of that field up to the mapper, but as it was throwing an error that mentioned a type mismatch, I wondered whether manually mapping it would help.

Sadly, I'm still getting this error even with the latest preview package (1.6.0-preview5). I'm trying to find a way to cut down this code to create a minimal reproduction. I have discovered that if I use `IEnumerable` instead of `List` in the mapping config I posted earlier, I get a different error:

 > System.ArgumentException: Expression of type 'System.Collections.Generic.IEnumerable\`1[MyApp.Dtos.ResultDto]' cannot be used for assignment to type 'System.Collections.Generic.List\`1[MyApp.Dtos.ResultDto]'

[agilemappererror2.txt](https://github.com/agileobjects/AgileMapper/files/3939021/agilemappererror2.txt)
